src/Essentials/src/Permissions/Permissions.ios.tvos.macos.cs

```diff
@@ -0,0 +1,102 @@
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Threading.Tasks;
+using Photos;
+
+namespace Microsoft.Maui.Essentials
+{
+	public static partial class Permissions
+	{
+		public partial class Photos : BasePlatformPermission
+		{
+			protected override Func<IEnumerable<string>> RequiredInfoPlistKeys =>
+				() => new string[] { "NSPhotoLibraryUsageDescription" };
+
+			public override Task<PermissionStatus> CheckStatusAsync()
+			{
+				EnsureDeclared();
+
+				return Task.FromResult(GetPhotoPermissionStatus(PHAccessLevel.ReadWrite));
+			}
+
+			public override async Task<PermissionStatus> RequestAsync()
+			{
+				EnsureDeclared();
+
+				var status = GetPhotoPermissionStatus(PHAccessLevel.ReadWrite);
+				if (status == PermissionStatus.Granted)
+				{
+					return status;
+				}
+#if __IOS__
+                else if (status == PermissionStatus.Limited)
+                {
+                    PhotosUI.PHPhotoLibrary_PhotosUISupport.PresentLimitedLibraryPicker(
+                        PHPhotoLibrary.SharedPhotoLibrary,
+                        Platform.GetCurrentUIViewController());
```

Comment:
This actually pops up a dialog that requires user input. If we return without waiting, then potentially the app will try access photos that the user is still selecting/deselecting.

There is an overload that takes a completion / awaitable:

```cs
await PHPhotoLibrary.SharedPhotoLibrary.PresentLimitedLibraryPickerAsync(Platform.GetCurrentUIViewController());
```